overlord/snapstate/snapstate.go

```diff
@@ -47,9 +47,13 @@ const (
 	// always be enforcing, even if the snap requests otherwise.
 	JailMode
 
-	// if we need flags for just SnapSetup it may be easier
-	// to start a new sequence from the other end with:
-	// 0x40000000 >> iota
+	// flags just for SnapSetup follow
+
+	// HasABA is a feature flag; doInstall sets it, and downstream handlers check for it to handle old SnapSetups
+	HasABA = 0x40000000 >> iota
```

Comment:
We only need the migration if we deal with tasks differently, which I sort of expected. For example, instead of had-candidate, we should probably have just old-candidate with an index, and always take the position in consideration if it exists. We cannot have clean logic for this without a migration.  It would break if non-revert cases that were working.